### PR TITLE
VZ-7750 Update Supported K8s versions in VPO to include 1.25

### PR DESCRIPTION
**Pull request by ayaranga**
Also, set Rancher chart helm value `global.cattle.psp.enabled` to false as PSP is not supported from K8s 1.25 onwards.

### CODE REVIEW

**Review by Abhijithang — APPROVED** [review]

File: platform-operator/helm_config/overrides/rancher-values.yaml [hunk 1]
```diff
@@ -1,2 +1,8 @@
-# Copyright (c) 2021, Oracle and/or its affiliates.
+# Copyright (c) 2021, 2023, Oracle and/or its affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+# Set global.cattle.psp.enabled to false as PSP not supported from K8s 1.25 onwards
+global:
+  cattle:
+    psp:
+      enabled: false
```
Comment (mgianatagh) [inline platform-operator/helm_config/overrides/rancher-values.yaml:8]: do we always want this set to false?  Or only when k8s 1.25 or higher?

Comment (mgianatagh) [reply platform-operator/helm_config/overrides/rancher-values.yaml:8]: decision was that PSP was deprecated earlier and ok to turn off for the versions of k8s we support.

**Review by mgianatagh — DISMISSED** [review]

**Review by mgianatagh — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go [hunk 2]
```diff
@@ -312,6 +320,47 @@ func appendImageOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.K
...
 
+// appendPSPEnabledOverrides appends overrides to disable PSP if the K8S version is 1.25 or above
+func appendPSPEnabledOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	version, err := getKubernetesClusterVersion()
+	if err != nil {
+		return kvs, ctx.Log().ErrorfNewErr("Failed to get the kubernetes version: %v", err)
+	}
+	k8sVersion, err := k8sversionutil.ParseSemantic(version)
+	if err != nil {
+		return nil, ctx.Log().ErrorfNewErr("couldn't parse Kubernetes version %q: %v", version, err)
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:331]: all errors should start with Failed ...

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go [hunk 3]
```diff
@@ -312,6 +320,47 @@ func appendImageOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.K
...
+		})
+	}
+	return kvs, nil
+}
+
+// getKubernetesVersion returns the version of Kubernetes cluster in which operator is deployed
+func getKubernetesVersion() (string, error) {
+	config, err := k8sutil.GetConfigFromController()
+	if err != nil {
+		return "", fmt.Errorf("error while getting kubernetes client config %v", err.Error())
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:348]: all errors should start with Failed ...

**Review by pfmackin — APPROVED** [review]

**Pull request merged**